proxmox/proxmox_instance.go

```diff
@@ -56,16 +57,124 @@ func (p *ProxMox) getNextID() string {
 
 // CreateInstance - Creates instance on Proxmox.
 func (p *ProxMox) CreateInstance(ctx *lepton.Context) error {
+
+	var err error
+
 	config := ctx.Config()
 
 	nextid := p.getNextID()
 
+	instanceName := config.RunConfig.InstanceName
+
+	archType := config.CloudConfig.Arch
+	machineType := config.CloudConfig.Machine
+	socketsNum := config.CloudConfig.Sockets
+	coresNum := config.CloudConfig.Cores
+	numaStr := strconv.FormatBool(config.CloudConfig.Numa)
+	memoryHmn := config.CloudConfig.Memory
 	imageName := config.CloudConfig.ImageName
+	storageName := config.CloudConfig.StorageName
+	isoStorageName := config.CloudConfig.IsoStorageName
+	bridgeName := config.CloudConfig.BridgeName
+	bridgeName0 := config.CloudConfig.BridgeName0
+	onbootStr := strconv.FormatBool(config.CloudConfig.Onboot)
+	protectionStr := strconv.FormatBool(config.CloudConfig.Protection)
+
+	// Check CloudConfig options
+
+	if archType == "" {
+		archType = "x86_64"
+	}
+
+	if socketsNum == 0 {
+		socketsNum = 1
+	}
+
+	socketsStr := strconv.FormatInt(int64(socketsNum), 10)
+
+	if coresNum == 0 {
+		coresNum = 1
+	}
+
+	coresStr := strconv.FormatInt(int64(coresNum), 10)
+
+	if numaStr == "true" {
+		numaStr = "1"
+	} else {
+		numaStr = "0"
+	}
+
+	if memoryHmn == "" {
+		memoryHmn = "512M"
+	}
+
+	memoryInt, err := units.RAMInBytes(memoryHmn)
```

Comment:
we're hoping to use less imports (and could use a deepclean on the ones that exist) but if possible it be good to just copy in whatever method you need here and plop it in here https://github.com/nanovms/ops/blob/master/lepton/helpers.go

Reply:
Ok, i`ll be move part of code (units.RAMInBytes) to lepton/helpers.go, and will remove import.